src/napari_builtins/_qt/features_table.py

```diff
@@ -121,10 +122,10 @@ def flags(self, index: QModelIndex) -> Qt.ItemFlag:
             return Qt.ItemFlag.ItemIsEnabled
 
         col = index.column()
-        # index is read-only
+        # index and layer_name columns are read-only
 
         flags = Qt.ItemFlag.ItemIsSelectable | Qt.ItemFlag.ItemIsEnabled
-        if col == 0:
+        if col == 0 or col == self.df.shape[1]:  # index or layer_name column
```

Comment:
Propagating something like this down the code would make it easier for me to read. Your thoughts?

```suggestion
        layer_name = self.df.shape[1]
        if col == 0 or col == layer_name:  # index or layer_name column
```

Reply:
This won't be needed once we move the layer column first.